scripts/dockerthinpool.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash -e
+
+# This script is meant to be chained after prepephemeral.sh, other scripts like
+# mountephemeral.sh can be used in a chain loaded fashion to then format and
+# mount the new scratchspace that lives on LVM as a logical volume.
+
+VGNAME=${VGNAME:-data}
+DOCKERPERCENT={$DOCKERPERCENT:-50}
+
+# 1% of the VG is already used by the thinpool meta, so the max number here is
+# 49 assuming you leave DOCKERPERCENT at 50. Start small and you can always
+# grow the logical volume later if necessary (lvextend)
+SCRATCHPERCENT=${SCRATCHPERCENT:-25}
+
+if [ -z "$SWARMYDIR" ]; then
+    echo "SWARMYDIR is not set, this script should be run from swarmy. Cowardly refusing to continue."
+    exit 1
+fi
+
+if [ ! -f "$SWARMYDIR/ephemeraldev" ]; then
+    echo "$SWARMYDIR/ephemeraldev does not exist, not sure where to format/mount"
+    exit 1
+fi
+
+read DEVICE < $SWARMYDIR/ephemeraldev
+
+if [ ! -b $DEVICE ]; then
+    echo "The device \"$DEVICE\" is not a block device. Cowardly refusing to continue"
+    exit 1
+fi
+
+if ! command -v lvm > /dev/null; then
+    echo "Installing required tools..."
+    yum install -y device-mapper-persistent-data lvm2
```

Comment:
Document dependencies for the scripts instead, and let cloud-init install them rather than doing it from the script.